### ISSUE TITLE
Enhance efficiency of release automation

### ISSUE THREAD

**Issue by aaemnnosttv** [google/site-kit-wp#9330]
## Feature Description

Now that the plugin release automation is working reliably, we can make some enhancements to make it a bit more efficient to run. Currently, doing a dry run and live run require invoking separate workflow runs which cannot run in parallel due to concurrency rules.

One simple enhancement we can do is to make the dry-run deployment part of the same workflow as a parallel job which doesn't depend on the input to the workflow run. By not running the dry-run in the same GH `environment`, we can also ensure no live deployment is possible during the dry-run.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The zips workflow for publishing the release, should be updated as follows
	*  A dry-run should always be performed, as a near-identical job to the current publish action without the ability to deploy the release (this should still be controlled by the current job. The dry-run shouldn't require a review/approval before running
	* The dry run input to the workflow should be removed, as each job should be fixed to be a dry-run or not

## Implementation Brief

* [ ] Update the zips workflow with a new job for the dry-run `dryrun-publish-to-wporg` which is the same as the existing `publish-to-wporg` with a few important differences
	* No defined `environment` (this excludes the secrets which include SVN credentials)
	* The deploy action is hardcoded to use with `dry-run: true`
	* Pass fake/no-op credentials in place of the SVN credentials since these are required by the script or it will return early
* [ ] Update the `publish-to-wporg` job to remove the `dry-run` option
	* Remove the `with` option
	* Also remove the corresponding `input` to the workflow as it's no longer needed
* Eventually this could use some kind of composite action to reduce duplication but it's not needed here as each job is rather small
* Finish and merge https://github.com/google/site-kit-wp/pull/9331

### Test Coverage

* No coverage to add/update

## QA Brief

* No QA required as this is only for release management.

## Changelog entry

* N/A.

**Comment by tofumatt** [comment]
IB ✅ 

I'm going to move this right to CR since there's a PR attached and I'm well-poised to review it as the other person now using this process 😅
